ledger-core/v2/virtual/testutils/slotmachine/slotmachine_wrapper.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package slotmachine
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gojuno/minimock/v3"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/instrumentation/inslogger"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log/global"
+	messageSenderAdapter "github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender/adapter"
+	"github.com/insolar/assured-ledger/ledger-core/v2/runner"
+	"github.com/insolar/assured-ledger/ledger-core/v2/runner/machine"
+	testUtilsCommon "github.com/insolar/assured-ledger/ledger-core/v2/testutils"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/synckit"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/integration/convlog"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/testutils"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/testutils/messagesender"
+)
+
+const (
+	runTimeout = 30 * time.Second
+)
+
+func NewWatchdog(duration time.Duration) func() bool {
+	timer := time.AfterFunc(duration, func() {
+		global.Fatalm(throw.FailHere("timeout"))
+	})
+	return timer.Stop
+}
+
+type ControlledSlotMachine struct {
+	t                *testing.T
+	ctx              context.Context
+	signal           *synckit.VersionedSignal
+	slotMachine      *smachine.SlotMachine
+	debugLogger      *testUtilsCommon.DebugMachineLogger
+	watchdogCancelFn func() bool
```

Comment:
Better to have a type here, as test may need to extend watchdog timeout / apply heartbeat